code/player/Player.cs

```diff
@@ -100,15 +117,51 @@ public override void StartTouch(Entity other)
             base.StartTouch(other);
         }
 
-        private void TickInspectPlayerCorpse()
+        private void TickAttemptInspectPlayerCorpse()
         {
-            // TODO: Handle role specific corpse inspecting.
-            if (!Input.Pressed(InputButton.Use))
+            if (IsClient)
             {
-                return;
+                if (InspectMenu.Instance?.IsShowing ?? false)
+                {
+                    // Menu is showing already, bail out.
+                    return;
+                }
             }
 
-            TraceResult trace = Trace.Ray(EyePos, EyePos + EyeRot.Forward * 80f)
+            if (IsServer)
+            {
+                using (Prediction.Off())
+                {
+                    To client = To.Single(this);
+                    PlayerCorpse playerCorpse = IsLookingAtPlayerCorpse();
+                    if (playerCorpse != null)
+                    {
+                        if (Input.Down(InputButton.Use) && !playerCorpse.IsIdentified)
+                        {
+                            playerCorpse.IsIdentified = true;
+                            Client playerCorpseInfo = playerCorpse.Player.GetClientOwner();
+                            ClientDisplayIdentifiedMessage(this.Controller.Client.SteamId,
+                                                        this.Controller.Client.Name,
+                                                        playerCorpseInfo.SteamId,
+                                                        playerCorpseInfo.Name,
+                                                        playerCorpse.Player.Role.ToString());
+                        }
+
+                        // Send the request to the player looking at the player corpse.
+                        // https://wiki.facepunch.com/sbox/RPCs#targetingplayers
+                        // TODO: Figure out why directly passing in just playerCorpse to "ClientOpenInspectMenu" makes playerCorpse.Player null.
+                        ClientOpenInspectMenu(client, playerCorpse.Player, playerCorpse.IsIdentified);
+                        return;
+                    }
+
+                    ClientCloseInspectMenu(client);
```

Comment:
Do we wanna use it globally or for better understanding with the prefix `this.`?

Reply:
I kind of like keeping all ClientRPCs static since by default they send a message to all clients.

Reply:
I think having Client prefix explains this enough, let me know if you disagree.